Notebook: Samikshya Pandey

##Task 1 

In [329]:
#importing libraries
import pandas as pd  
import numpy as np   

In [330]:
data = pd.read_csv("chipotle.tsv" ,sep='\t')   #loading the dataset

In [331]:
data.head() 
data.info()    

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   order_id            4622 non-null   int64 
 1   quantity            4622 non-null   int64 
 2   item_name           4622 non-null   object
 3   choice_description  3376 non-null   object
 4   item_price          4622 non-null   object
dtypes: int64(2), object(3)
memory usage: 180.7+ KB


Which was the most ordered item? 


In [332]:
most_ordered = data.groupby(["item_name"])["quantity"].sum()
most_ordered.head(10)
most_ordered.sort_values(ascending = False).head(1)  


item_name
Chicken Bowl    761
Name: quantity, dtype: int64

The most ordered item is chicken bowl 

Question 2. How many items were ordered in total? 

In [333]:
#Calculating unique items
data_total = data["quantity"].sum()   
print(data_total)   



4972


We can see that 4972 items were ordered in total. 

How much was the revenue for the period in the dataset?

In [334]:
#Revunue = qunatity * item_price 
data[["item_price", "quantity"]].head(5) #All items in the item price has a $ so we cnanot do calculation yet.  
#Removing the $ sign


item_price  quantity
0     $2.39          1
1     $3.39          1
2     $3.39          1
3     $2.39          1
4    $16.98          2

In [335]:
data["item_price"] = data["item_price"].str.replace('$' , '')  #Removing the $ for calculation
#data.head() 
data["item_price"] = data["item_price"].astype(str).astype(float)  #convert the string to float
data["quantity"] = data["quantity"].astype(float) #convery quantity to float to make calculation easier




In [336]:
data.info() #checking if data type is similar

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 5 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   order_id            4622 non-null   int64  
 1   quantity            4622 non-null   float64
 2   item_name           4622 non-null   object 
 3   choice_description  3376 non-null   object 
 4   item_price          4622 non-null   float64
dtypes: float64(2), int64(1), object(2)
memory usage: 180.7+ KB


In [337]:
data['Revenue'] = data["quantity"] * data["item_price"]  #Revenue = quantitiy * item price
data['Revenue'].sum()




39237.02

we can see that the total revenue collected is 39237.02

How many orders were made? Order = sum total 

In [338]:
#We can look at how many ordered were made based on unique order_id value so 
sum_order = data["order_id"].nunique()      
sum_order  

1834

There were 1834 orderes made. 

How many times was a Veggie Salad Bowl ordered?

In [339]:
veggie_bowl = data[data["item_name"] == "Veggie Salad Bowl"]["quantity"].sum() 
veggie_bowl 


18.0

So, we can see that the veggie bowl salad was ordered 18 times 

How many times did someone order more than one Canned Soda?

In [340]:
can_data = data[(data["item_name"] == "Canned Soda")  & (data["quantity"] > 1)] 
print(can_data["quantity"].count())  #Printing the quantity of the canned soda order b


20


There were 20 times where canned soda was ordered more than one time. 

Find the orders that cost more than $100

In [341]:
data["Total_cost"] = data["item_price"] * data["quantity"] 
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4622 entries, 0 to 4621
Data columns (total 7 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   order_id            4622 non-null   int64  
 1   quantity            4622 non-null   float64
 2   item_name           4622 non-null   object 
 3   choice_description  3376 non-null   object 
 4   item_price          4622 non-null   float64
 5   Revenue             4622 non-null   float64
 6   Total_cost          4622 non-null   float64
dtypes: float64(4), int64(1), object(2)
memory usage: 252.9+ KB


In [342]:
total = data[["Total_cost", "order_id"]].groupby(["order_id"]).sum()  #groupby.orderid groupby order id did the sum
total_100 = total[total["Total_cost"] > 100] 
print(total_100)   #printing the order_id whose total cose was greater than 100. 

          Total_cost
order_id            
205           109.90
491           102.00
511           315.29
691           118.25
759           124.10
926           205.25
1398          105.75
1443         1074.24
1449          164.31
1454          190.24
1483          139.00
1559          246.00
1660          222.75
1764          101.55
1786          197.70


## Task 2 

In [343]:
data_student = pd.read_csv("student-mat.txt" , sep = ',') #Loading the dataset.
data_student.head()

school sex  age address famsize Pstatus  Medu  Fedu     Mjob      Fjob  ...  \
0     GP   F   18       U     GT3       A     4     4  at_home   teacher  ...   
1     GP   F   17       U     GT3       T     1     1  at_home     other  ...   
2     GP   F   15       U     LE3       T     1     1  at_home     other  ...   
3     GP   F   15       U     GT3       T     4     2   health  services  ...   
4     GP   F   16       U     GT3       T     3     3    other     other  ...   

  famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0      4        3      4     1     1      3        6   5   6   6  
1      5        3      3     1     1      3        4   5   5   6  
2      4        3      2     2     3      3       10   7   8  10  
3      3        2      2     1     1      5        2  15  14  15  
4      4        3      2     1     2      5        4   6  10  10  

[5 rows x 33 columns]

Slicing the dataframe from ‘school’ until ‘guardian’ and save it into new dataframe

In [344]:
df_student2 = data_student.loc[:,'school':'guardian'] #Creating a new dataframe
df_student2

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     reason guardian  
0    course   mother  
1    course   father  
2     other   mother  
3      home   mother  
4      home   father  
..      ...      ...  
390  course    other  
391  course   mother  
392  course    other  
393  course   mother  
394  course   father  

[395 rows x 12 columns]

In [345]:
f = lambda x: x.str.title()  #Creating a function that converts the first letter into a capital
df_student2[["Mjob", "Fjob"]] = df_student2[["Mjob", "Fjob"]].apply(f)
df_student2.head(10)

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4   At_Home   Teacher   
1     GP   F   17       U     GT3       T     1     1   At_Home     Other   
2     GP   F   15       U     LE3       T     1     1   At_Home     Other   
3     GP   F   15       U     GT3       T     4     2    Health  Services   
4     GP   F   16       U     GT3       T     3     3     Other     Other   
5     GP   M   16       U     LE3       T     4     3  Services     Other   
6     GP   M   16       U     LE3       T     2     2     Other     Other   
7     GP   F   17       U     GT3       A     4     4     Other   Teacher   
8     GP   M   15       U     LE3       A     3     2  Services     Other   
9     GP   M   15       U     GT3       T     3     4     Other     Other   

       reason guardian  
0      course   mother  
1      course   father  
2       other   mother  
3        home   mother  
4        home   father  
5  reputation   mother  
6        home   mother  
7        home   mother  
8        home   mother  
9        home   mother

Add a new column "legal_drinker" which has a boolean value. If age is below 21, the value is 
false; otherwise, the value is true. 

In [346]:
df_student2["legal_drinker"] = np.where(df_student2["age"] < 21, False, True)  #pass true for age greater or bigger than 21
df_student2.head(10)


school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4   At_Home   Teacher   
1     GP   F   17       U     GT3       T     1     1   At_Home     Other   
2     GP   F   15       U     LE3       T     1     1   At_Home     Other   
3     GP   F   15       U     GT3       T     4     2    Health  Services   
4     GP   F   16       U     GT3       T     3     3     Other     Other   
5     GP   M   16       U     LE3       T     4     3  Services     Other   
6     GP   M   16       U     LE3       T     2     2     Other     Other   
7     GP   F   17       U     GT3       A     4     4     Other   Teacher   
8     GP   M   15       U     LE3       A     3     2  Services     Other   
9     GP   M   15       U     GT3       T     3     4     Other     Other   

       reason guardian  legal_drinker  
0      course   mother          False  
1      course   father          False  
2       other   mother          False  
3        home   mother          False  
4        home   father          False  
5  reputation   mother          False  
6        home   mother          False  
7        home   mother          False  
8        home   mother          False  
9        home   mother          False